### PR TITLE
Ce/static incentive

### PR DESCRIPTION
**Pull request by calellowitz**
@Rohit25negi @emord 
This adds the columns to the incentive report table to make sure that we no longer have to do any calculations in python (I did division in there, since i don't think that can change, but would be open to pulling that into the db as well). Once this is merged I will backfill the new columns

I will take another pass at this for some cleanup but wanted to get tests running and any early feedback

### CODE REVIEW

**Comment by calellowitz** [comment]
made some changes based on new requirements but this is now ready for review. The big change is that we now pregenerate all of these reports, and then fetch the pregenerated version, instead of creating the excel report each time the report is requested.
code buddies @snopoke @mkangia

File: custom/icds_reports/reports/incentive.py [hunk 1]
```diff
@@ -65,12 +70,16 @@ def _format_infrastructure_data(data):
...
             else:
-                if self.month.year < 2019 or (self.month.year == 2019 and self.month.month < 3):
-                    func = int
+                if self.beta:
+                    visit_denominator = row_data['visit_denominator']
                 else:
-                    func = round
-                home_visit_percent = row['valid_visits'] / func(row['expected_visits']) if \
-                    func(row['expected_visits']) else 1
+                    if self.month.year < 2019 or (self.month.year == 2019 and self.month.month < 3):
```
Comment (snopoke) [inline custom/icds_reports/reports/incentive.py]: a comment here would be helpful

File: custom/icds_reports/reports/incentive.py [hunk 2]
```diff
@@ -73,12 +78,16 @@ def _format_infrastructure_data(data):
...
+                        func = int
+                    else:
+                        func = round
```
[the remark this answers is not in the document]
Comment (snopoke) [reply custom/icds_reports/reports/incentive.py]: py 3

File: custom/icds_reports/reports/incentive.py [hunk 3]
```diff
@@ -89,11 +98,16 @@ def _format_infrastructure_data(data):
...
                 else:
                     num_open = row['awc_num_open']
-                    awh_eligible_for_incentive = 'Yes' if int(num_open) >= 21 else 'No'
+
                 weighing_efficiency = '{:.2%}'.format(weighing_efficiency_percent)
-                eligible_for_incentive = 'Yes' if \
-                    weighing_efficiency_percent >= 0.6 and home_visit_percent >= 0.6 else 'No'
-                no_visits = row['valid_visits'] == 0 and row['expected_visits'] == 0
+                if self.beta:
+                    awh_eligibile_for_incentive = 'Yes' if row['awh_eligible'] else 'No'
```
[the remark this answers is not in the document]
Comment (snopoke) [reply custom/icds_reports/reports/incentive.py]: This is spelled differently from before:
```suggestion
                    awh_eligible_for_incentive = 'Yes' if row['awh_eligible'] else 'No'
```

File: custom/icds_reports/tasks.py [hunk 4]
```diff
@@ -1168,11 +1157,49 @@ def build_incentive_report(agg_date=None):
...
+                 .select_related('parent__parent', 'location_type'))
     if agg_date is None:
         current_month = date.today().replace(day=1)
         agg_date = current_month - relativedelta(months=1)
     for state in state_ids:
         AWWIncentiveReport.aggregate(state, agg_date)
+    for file_format in ['xlsx', 'csv']:
+        for location in locations:
+            if location.location_type.name == 'state':
+                build_incentive_files.delay(location.location_id, agg_date, file_format, 1, location)
```
Comment (snopoke) [inline custom/icds_reports/tasks.py]: This should be passing the full location and not just `location.location_id`

File: custom/icds_reports/tasks.py [hunk 5]
```diff
@@ -1168,11 +1157,49 @@ def build_incentive_report(agg_date=None):
...
+            elif location.location_type.name == 'district':
+                build_incentive_files.delay(location, agg_date, file_format, 2, location.parent, location)
+            else:
+                build_incentive_files.delay(location, agg_date, file_format, 3, location.parent.parent, location.parent, location)
```
Comment (snopoke) [inline custom/icds_reports/tasks.py]: This appears to have too many args. I don't think the last `location` arg should be there.

File: custom/icds_reports/tasks.py [hunk 6]
```diff
@@ -1168,11 +1157,49 @@ def build_incentive_report(agg_date=None):
...
+def build_incentive_files(location, month, file_format, aggregation_level, state, district=None, beta=False):
+    data_type = 'AWW_Performance'
+    excel_data = IncentiveReport(
+        location=location.location_id,
+        month=month,
+        aggregation_level=aggregation_level,
+        beta=beta
+    ).get_excel_data()
+    state = state.name
+    district = district.name if aggregation_level >= 2 else None
```
Comment (snopoke) [inline custom/icds_reports/tasks.py]: or `district_name = district.name if district else None`

Also FYI you are overriding the var names here for `state` and `district`

**Review by snopoke — CHANGES_REQUESTED** [review]

File: custom/icds_reports/reports/incentive.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (calellowitz) [reply custom/icds_reports/reports/incentive.py]: This is current code, and will be removed once QA passes on these changes

File: custom/icds_reports/reports/incentive.py (hunk 1 above)
Comment (calellowitz) [reply custom/icds_reports/reports/incentive.py]: Same comment as above. we are removing this hopefully within the week but can add a comment in the meantime if necessary

**Review by snopoke — APPROVED** [review]

File: custom/icds_reports/reports/incentive.py [hunk 7]
```diff
@@ -17,23 +17,27 @@ def get_excel_data(self):
         def _format_infrastructure_data(data):
             return data if data else DATA_NOT_ENTERED
 
+        model = AWWIncentiveReport if self.beta else AWWIncentiveReportMonthly
```
[the remark this answers is not in the document]
Comment (snopoke) [reply custom/icds_reports/reports/incentive.py]: missing import

**Review by snopoke — CHANGES_REQUESTED** [review]

File: custom/icds_reports/reports/incentive.py (hunk 7 above)
[the remark this answers is not in the document]
Comment (snopoke) [reply custom/icds_reports/reports/incentive.py]: strange, I can see the import when I check out the branch locally

**Review by snopoke — APPROVED** [review]

**Pull request merged**